### PR TITLE
Parse more fields for Copilot list seats API

### PR DESCRIPTION
**Pull request by timweri**
Changes in order of importance:
1. I need more parsed fields in the List Copilot Seats API to tell when seats are team-based and when seats have already been queued for removal.
2. Update datetime deserializers
3. Remove an unused line flagged by the compiler.

### CODE REVIEW

File: plaid-stl/src/npm/shared_structs/mod.rs [hunk 1]
```diff
@@ -169,7 +168,7 @@ pub struct NpmToken {
     pub token_name: Option<String>,
     pub token_type: Option<String>,
     // Deserialize the date field from the ISO 8601 format
-    #[serde(default, deserialize_with = "deserialize_option_timestamp")]
+    #[serde(with = "chrono::serde::ts_seconds_option")]
```
Comment (timweri) [inline plaid-stl/src/npm/shared_structs/mod.rs:171]: @michelemin  `chrono` has a built in function for this. Do you bless this or should I leave your portion as is?

File: plaid-stl/src/github/mod.rs [hunk 2]
```diff
@@ -40,20 +40,31 @@ pub struct CopilotAssignee {
...
-    pub created_at: Option<DateTime<Utc>>,
-    // Deserialize the date field from the ISO 8601 format
-    #[serde(default, deserialize_with = "deserialize_option_timestamp")]
-    pub updated_at: Option<DateTime<Utc>>,
+    // Deserialize the datetime field from the ISO 8601 format
+    #[serde(with = "chrono::serde::ts_seconds")]
+    pub created_at: DateTime<Utc>,
+    // Deserialize the datetime field from the ISO 8601 format
+    #[serde(with = "chrono::serde::ts_seconds")]
+    pub updated_at: DateTime<Utc>,
```
Comment (timweri) [inline plaid-stl/src/github/mod.rs:67]: According to the API schema, `created_at` and `updated_at` are not null.

File: plaid-stl/src/datetime.rs [hunk 3]
```diff
@@ -1,18 +1,64 @@
...
+    // throwing parsing errors
+    if s.is_empty() {
+        return Ok(None)
+    }
+
+    Ok(Some(
+        DateTime::parse_from_rfc3339(&s)
+            .map_err(serde::de::Error::custom)?
+            .with_timezone(&Utc)
+    ))
```
Comment (timweri) [inline plaid-stl/src/datetime.rs:40]: @michelemin  If the string is empty, I return None. Before, if the string is not empty and parsing fails, the deserializer returns None, which I think is misleading. Now, if parsing fails on an non-empty string, the deserializer will throw an error.

**Review by michelemin — APPROVED** [review]
LGTM

File: plaid-stl/src/datetime.rs [hunk 4]
```diff
@@ -1,18 +1,64 @@
-use chrono::{DateTime, Utc};
+use chrono::{NaiveDate, Utc, DateTime};
 use serde::{Deserialize, Deserializer};
 
-// Custom deserializer for an optional DateTime<Utc>
-pub fn deserialize_option_timestamp<'de, D>(deserializer: D) -> Result<Option<DateTime<Utc>>, D::Error>
+// Custom deserializer from a ISO 8601 string to a DateTime<Utc>
+pub fn deserialize_rfc3339_timestamp<'de, D>(deserializer: D) -> Result<DateTime<Utc>, D::Error>
```
Comment (obelisk) [inline plaid-stl/src/datetime.rs:5]: Interesting choice in the comment to refer to it as ISO 8601 and the function RFC3339?

**Review by obelisk — APPROVED** [review]

File: plaid-stl/src/datetime.rs (hunk 4 above)
Comment (timweri) [reply plaid-stl/src/datetime.rs:5]: :P I want to call it ISO 8601 but the chrono function uses RFC 3339

**Pull request merged**